### PR TITLE
test (salsah): add running of integration tests on travis-ci

### PR DESCRIPTION
**Pull request by subotic**
### Summary

 - added webapi integration test execution on Travis-CI
 - added headless browser mode
 - updated chromedriver to the newest version
 - added a workaround for closing the alert window to work in headless mode
 - salsah gui, as one would expect, is now served under:
   - `http://localhost:3335`
   - `http://localhost:3335/`
   - `http://localhost:3335/index.html`
 - updated `README.md` to reflect that the chromedriver can be found inside `salsah/lib/chromedriver`
 - add travis-ci stage for building plain vanilla docker images of `webapi` and `salsah`, and publishing them on docker hub.
 - upgraded `salsah` to the newest `akka-http` version (not using spray anymore).

To run salsah tests in headless mode: ``$ sbt headless:test``.
To run salsah tests as before: ``$ sbt test``.

To run integration tests with graphdb: ``$ sbt it:test``
To run integration tests with fuseki: ``$ sbt fuseki-it:test``

### CODE REVIEW

**Comment by subotic** [comment]
@benjamingeer I have also added running of integration tests on Travis-CI. There seems to be a problem (I think) with Sipi: https://travis-ci.org/dhlab-basel/Knora/jobs/241272169#L1370 

I get the same error when I run the integration tests locally. Can you maybe take a look?

**Comment by subotic** [comment]
I give up (for now) on running the salsah tests on travis. At least we have the integration tests.

**Comment by subotic** [comment]
The problem with the `sipi` docker image is fixed and the travis build is green again. Whenever you have time.

File: salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala [hunk 1]
```diff
@@ -88,7 +96,7 @@ class SalsahPage {
       * @param password user's password
       * @param fullName user's full name
       */
-    def doLogin(email: String, password: String, fullName: String): Unit = {
+    def doLogin(email: String, password: String, fullName: String, _page: SalsahPage): Unit = {
```
Comment (benjamingeer) [inline salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala]: Why does `page.doLogin` need to get passed a reference to `page`? Isn't that just `this`?

File: salsah/src/test/scala/org.knora.salsah/browser/SearchAndEditSpec.scala [hunk 2]
```diff
@@ -93,25 +85,25 @@ class SearchAndEditSpec extends SalsahSpec {
...
             page.getPageTitle should be ("System for Annotation and Linkage of Sources in Arts and Humanities")
 
         }
 
         "log in as an Incunabula project user" in {
 
-            page.load()
-            page.doLogin(email = incunabulaUserEmail, password = testPassword, fullName = incunabulaUserFullName)
+            page.open()
+            page.doLogin(email = incunabulaUserEmail, password = testPassword, fullName = incunabulaUserFullName, page)
```
Comment (benjamingeer) [inline salsah/src/test/scala/org.knora.salsah/browser/SearchAndEditSpec.scala]: Why do you pass `page` to `page.doLogin`? Wouldn't that just be `this` inside the method?

File: salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala (hunk 1 above)
Comment (subotic) [inline salsah/src/test/scala/org.knora.salsah/browser/SalsahPage.scala]: It doesn't need it. Just wanted to see if you could catch the error ;-)

File: salsah/src/test/scala/org.knora.salsah/browser/SearchAndEditSpec.scala (hunk 2 above)
Comment (subotic) [inline salsah/src/test/scala/org.knora.salsah/browser/SearchAndEditSpec.scala]: It doesn't need it. Just wanted to see if you could catch the error ;-)

**Comment by subotic** [comment]
This is ready for review whenever you have time.

**Comment by benjamingeer** [comment]
The Travis builds are failing here, but I don't understand why.

**Comment by subotic** [comment]
The integration test is using sipi from the develop branch. The latest merge into the sipi `develop` branch has broken the integration tests. Should work again after https://github.com/dhlab-basel/Sipi/pull/174 is ready.

**Comment by benjamingeer** [comment]
OK, looks fine to me, then, thank you and please merge.

**Comment by subotic** [comment]
Thanks :-)

**Pull request merged**